rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/IllegalOperationChecker.java

```diff
@@ -0,0 +1,142 @@
+                final int characteristicProperties = characteristic.getProperties();
+
+                if ((characteristicProperties & neededProperties) == 0) {
+                    int[] possibleProperties = getPossibleProperties();
+                    String message = String.format(
+                            Locale.getDefault(),
+                            "Characteristic %s supports properties:[%s] (%d) does not have any property matching [%s] (%d)",
+                            characteristic.getUuid(),
+                            propertiesIntToString(characteristicProperties, possibleProperties),
+                            characteristicProperties,
+                            propertiesIntToString(neededProperties, possibleProperties),
+                            neededProperties
+                    );
+                    resultHandler.handleMismatchData(message, characteristic.getUuid(), characteristicProperties, neededProperties);
+                }
+                return null;
+            }
+        });
+    }
+
+    @NonNull
+    private String propertiesIntToString(int property, int[] possibleProperties) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(" ");
+        for (int possibleProperty : possibleProperties) {
+            if (propertiesIntContains(property, possibleProperty)) {
+                builder.append(propertyToString(possibleProperty));
+                builder.append(" ");
+            }
+        }
+        return builder.toString();
+    }
+
+    private int[] getPossibleProperties() {
+        int[] propertyDictionary = new int[7];
+        propertyDictionary[0] = propertyBroadcast;
+        propertyDictionary[1] = propertyRead;
+        propertyDictionary[2] = propertyWriteNoResponse;
+        propertyDictionary[3] = propertyWrite;
+        propertyDictionary[4] = propertyNotify;
+        propertyDictionary[5] = propertyIndicate;
+        propertyDictionary[6] = propertySignedWrite;
+        return propertyDictionary;
+    }
+
+    private static boolean propertiesIntContains(int properties, int property) {
+        return (properties & property) != 0;
+    }
+
+    @NonNull
+    private String propertyToString(int property) {
+        if (property == propertyRead) {
+            return "READ";
+        } else if (property == propertyWrite) {
+            return "WRITE";
+        } else if (property == propertyWriteNoResponse) {
+            return "WRITE_NO_RESPONSE";
+        } else if (property == propertySignedWrite) {
+            return "SIGNED_WRITE";
+        } else if (property == propertyIndicate) {
+            return "INDICATE";
+        } else if (property == propertyBroadcast) {
+            return "BROADCAST";
+        } else if (property == propertyNotify) {
+            return "NOTIFY";
+        } else {
+            RxBleLog.e("Unknown property specified");
+            throw new IllegalArgumentException("Unknown property specified");
```

Comment:
Is it documented that the method may throw IllegalArgumentException? It will be passed to the public API.